cmd/otelcontribcol/components.go

```diff
@@ -70,18 +60,7 @@ func components() (config.Factories, error) {
 		errs = append(errs, err)
 	}
 
-	receivers := []component.ReceiverFactoryBase{
-		&collectdreceiver.Factory{},
-		&sapmreceiver.Factory{},
-		&signalfxreceiver.Factory{},
-		&carbonreceiver.Factory{},
-		&wavefrontreceiver.Factory{},
-		&redisreceiver.Factory{},
-		&kubeletstatsreceiver.Factory{},
-		&simpleprometheusreceiver.Factory{},
-		&k8sclusterreceiver.Factory{},
-		&receivercreator.Factory{},
-	}
+	receivers := allReceivers()
```

Comment:
To avoid having to duplicate all the receivers in both sub files, how about making this something like:
```go
receivers := []component.ReceiverFactoryBase{
	...
}
receivers = append(receivers, additionalReceivers()...)
```

Reply:
Agreed. Renamed the function to `additionalReceivers()` and fix it to return `cudareceiver.Factory{}` in the case of `gpu` flag is on and empty list in the case the flag is off.